### PR TITLE
(#4204) - Ensure we respect revs_limit on new_edits writes

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
So @nolanlawson you are probably not going to love this, all new_writes edits have been hitting the database directly so they have all been avoiding `merge` which makes the writes faster but it can lead to us having unbounded revisions stored. This is likely going to cause a bit of a slowdown however its the correct behaviour now.

File: lib/deps/docs/processDocs.js [hunk 1]
```diff
@@ -84,6 +84,9 @@ function processDocs(docInfos, api, fetchedDocs, tx, results, writeDoc, opts,
         updateDoc(fetchedDocs.get(id), currentDoc, results,
           resultsIdx, docWritten, writeDoc, newEdits);
       } else {
+        // Ensure stemming applies to new writes as well
+        var merged =  merge.merge([], currentDoc.metadata.rev_tree[0], 1000);
```
Comment (nolanlawson) [inline lib/deps/docs/processDocs.js]: ^ should probably extract that 1000 to a shared constant somewhere.

**Comment by nolanlawson** [comment]
+1 when green. Also I'll look to see if there's a way to speed this up by traversing the tree to detect whether or not we're at <1000 revs.

File: lib/deps/docs/processDocs.js (hunk 1 above)
Comment (daleharvey) [inline lib/deps/docs/processDocs.js]: Yeh I was planning to do https://github.com/pouchdb/pouchdb/issues/2839 at some point soon

**Comment by daleharvey** [comment]
Green, merged cheers - https://github.com/pouchdb/pouchdb/commit/13e9c18dce44d90366d902a2e5ba2f34184d5887

**Pull request closed without merging**